### ISSUE TITLE
[1.11.x+]育成システムについて / How about training system?

### ISSUE THREAD

**Issue by Verclene** [Verclene/LittleMaidReengaged#95]
どうもVercleneです.
考えがイマイチ纏まらないというか,方向性が定まらなさすぎて正直開発のモチベーションも上がらないので,ここいらで制作者自ら議論用のトピックを投げてみます.

その愛くるしさと便利さが人気を集めてやまないメイドさんですが,ここんとこ~~これでも~~ちょいと便利すぎるように感じています.端的に言うと,はじめからずっとできることが変わらないのが~~これでもゲーマーの端くれとして~~退屈かな,と.
一応,レベリングもどきの要素は実装していて,レベルが上がると粗品がもらえるボーナスがあったりしますが,これの拡張がてらにも何か作りたかった感あります.

そこで今私の頭の中で考えているのはズバリ"メイドさん育成システム"です.
1. レベルが上がる等のトリガーで技術ポイントみたいなものがメイドさんごとにもらえる
2. これを振り分けることで専門職が解放されていく
3. 全てにスキルを振り分けられるわけではない
現代のオンラインゲームではよくあるヤツです.~~ぷそにばっかりやってるけど~~
メイドさんを育てるというやりこみ要素と,メイドさんごとに個性を持たせる特化要素をもたせるのを,一応狙いとしています.

問題としては,
* 専門職を用意できるほど今のモードが多くない
* 本家メイドさんの実装を大きく覆すことになる
* レベリングめんどい
* いわゆる"エンドコンテンツ"はどの程度にする?

などがあげられますが,この辺是非メイド愛好家の皆様の目に留まったら,何か書いていってください.

ただしネトゲシステムなんぞクソクラエとかいう個人の思想だけ書いていくのはおやめください.そういうのはよそでやれ.

[English text](https://github.com/Verclene/LittleMaidReengaged/issues/95#issuecomment-292792967) ~~will be added later.~~ Thanks for summarizing

**Comment by Verclene** [comment]
差しあたっては"採掘"や"収集"とか"栽培"とかの分野別にメイドさんを特化できるようにして,それぞれに技術ポイント獲得用アイテムを用意するとかどうでしょうか.

**Comment by Shionsan** [comment]
LittleMaidMob: The RPG?

What are we talking about here:
- killing mobs, getting level up, spending points (common RPG style)?
- or upgrading skills by using them (skyrim style)?

And what is there to upgrade besides attack power, attack speed, running speed and protection values?

Do you mean to add something like skills for maids ("Using shield", "Dual wield", "Farmer's aura", "Faster mining", "Minerals detection", "Magic!?" etc)? Or skills for a player ("Max number of maids in group", strategical commands like "attack", "defend", some player buffs based on what maids are in group)?

**Comment by Verclene** [comment]
>Do you mean to add something like skills for maids ("Using shield", "Dual wield", "Farmer's aura", "Faster mining", "Minerals detection", "Magic!?" etc)?]

Exactly. Maids can be more useful, but masters have to train them beforehand. Now they can do many tasks without anything like that, so I have felt that they are too excellent to be employed from early in the game.

>Or skills for a player ("Max number of maids in group", strategical commands like "attack", "defend", some player buffs based on what maids are in group)?

The maids have performed independently to support their masters(even in earlier LMM versions), so I hesitate about adding skills which enable masters to command maids in detail. (There was an addon by which you can do that for old LMM, of course.)

**Comment by Verclene** [comment]
And there's another problem: "Who gets the skill points?": Each maid has to gain points or only master has to?

**CASE 1 (Master gets points)**: Each maid can allot points to different skills up to her master's point. (ex)If the master has 50 pts, a maid can allot up to 50 pts. The master's points are not consumed for apportioning.
**CASE 2 (Master gets points)**: Each maid can allot points to skills, and also consume the master's points. (ex)If the master has 50 pts and a maid allots 20 pts, the master's points become 30.
**CASE 3 (Maid gets points)**: Each maid gets points and allots them to skills.

**Comment by Shionsan** [comment]
In CASE 1 maid won't be able to become "omnipotent", and "training" can become pain really fast. Do not forget, that player's level (points) are spendable and will reset to zero on death, so there won't be any training if your level drops lower that of your maid'. And I don't really see why maid shouldn't be better/stronger than player.

CASE 2 doesn't look bad, but it will look more like "enchanting/teaching" than "training", as you would literally use player's exp, which comes from doing all the stuff by yourself. But it would work if all the exp maids gain for doing their jobs were to go to the player. Have to balance the skills cost too. It's not hard to get first 5 levels. But if you were to need 30+...

CASE 3 seems to be the best option, but... how many ways are there for maid to level up (gain points)? Won't it be tedious or plain boring? Won't it force player to spend points only into combat skills first?

And if we talk about RPGs, will there be a ressurection system? I mean, there are TNT maids, friendly fire (fellow maid archers XD), ender dragons and that 3-headed ghost dude, other crazy players, mods with new powerfull enemies who could kill a full equipped maid in 2 hits and LAVA (maids don't seem to be good in navigating caves just yet). Maid's death is permament right now. So if you want to make them into something one have to put time into, you'd better make it so all won't be lost easily to bugs or something else outside of player's control. Might also need to make maids teleport-friendly (recall ability). So you won't have them lost during your travels. Unless you're going to add that "maid in distress" mode I suggested last year.

And about "omnipotency": will maid be able to learn everything? Will there be a max level? Max level of skills? Max level of stats (damage, defence, etc.)? Or you just thinking about complex ways to enable maids "professions"?

**Comment by Verclene** [comment]
>that player's level (points) are spendable

I won't use the vanilla experience points. One or some additional status may be used.

>will there be a ressurection system?

Already exists, but currently it requires so much sugar.

**Comment by Shionsan** [comment]
With that in mind I'd say CASE 3 is the right choice. Especially if maids would gain points not only by fighting, but also by doing other jobs.

And I think there is a need to use books in some way. For example: a new mode, when you give maid a book, she just sits there (like zabuton mod) and gain exp by self-teaching (safe method, should take longer). Or use bookshelves as exp multiplier. If one is in the vicinity - the maid would gain like 1% more exp per block from whatever she is doing. That idea kinda makes me want to build a library or maybe even a school XD
